content/en/blog/2019/monitoring-external-service-traffic/index.md

```diff
@@ -0,0 +1,379 @@
+        "destination_principal": "unknown",
+        "destination_service": "httpbin.org",
+        "destination_service_name": "BlackHoleCluster",
+        "destination_service_namespace": "unknown",
+        "destination_version": "unknown",
+        "destination_workload": "unknown",
+        "destination_workload_namespace": "unknown",
+        "instance": "100.96.2.183:42422",
+        "job": "istio-mesh",
+        "permissive_response_code": "none",
+        "permissive_response_policyid": "none",
+        "reporter": "source",
+        "request_protocol": "http",
+        "response_code": "502",
+        "response_flags": "-",
+        "source_app": "sleep",
+        "source_principal": "unknown",
+        "source_version": "unknown",
+        "source_workload": "sleep",
+        "source_workload_namespace": "default"
+      },
+      "value": [
+        1567034251.717,
+        "1"
+      ]
+    }
+  {{< /text >}}
+
+  Note the `destination_service_name` label is set to BlackHoleCluster and the
+  `destination_service` to the host name of the external service. The response
+  code should always be `502` in this case.
+
+* TCP proxy virtual listener - If the external service port doesn't map to any
+  HTTP based service ports within the cluster, this listener is invoked and
+  `istio_tcp_connections_closed_total` is the metric that will be increased:
+
+  {{< text json >}}
+    {
+      "metric": {
+        "__name__": "istio_tcp_connections_closed_total",
+        "connection_security_policy": "unknown",
+        "destination_app": "unknown",
+        "destination_ip": "52.22.188.80",
+        "destination_principal": "unknown",
+        "destination_service": "unknown",
+        "destination_service_name": "BlackHoleCluster",
+        "destination_service_namespace": "unknown",
+        "destination_version": "unknown",
+        "destination_workload": "unknown",
+        "destination_workload_namespace": "unknown",
+        "instance": "100.96.2.183:42422",
+        "job": "istio-mesh",
+        "reporter": "source",
+        "response_flags": "-",
+        "source_app": "sleep",
+        "source_principal": "unknown",
+        "source_version": "unknown",
+        "source_workload": "sleep",
+        "source_workload_namespace": "default"
+      },
+      "value": [
+        1567034481.03,
+        "1"
+      ]
+    }
+  {{< /text >}}
+
+  Note the `destination_ip` label represents the IP address of the external
+  service and the `destination_service_name` is set to BlackHoleCluster
+  to indicate that this traffic was blocked by the mesh. Is is interesting to
+  note that for the BlackHole cluster case, other TCP related metrics like
+  `istio_tcp_connections_opened_total` are not increased as there's no
+  connection that is ever established.
+
+Monitoring these metrics can help operators easily understand all the external
+services consumed by the applications in their cluster.
+
+### References
```

Comment:
Is this strictly necessary? We generally don't want customers redirected to GitHub if we can avoid it.